src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -400,43 +450,49 @@ protected virtual void Dispose(bool finalize)
 
         #region Host API
 
+        private static Solution CheckAndAddProject(Solution newSolution, ProjectInfo project)
+        {
+            CheckProjectIsNotInSolution(newSolution, project.Id);
+            newSolution = newSolution.AddProject(project);
```

Comment:
Inline since you can get rid of the assignment entirely?

Reply:
done.